src/rules/no-missing-atx-heading-space.js

```diff
@@ -10,17 +10,46 @@
 /**
  * @import { MarkdownRuleDefinition } from "../types.js";
  * @typedef {"missingSpace"} NoMissingAtxHeadingSpaceMessageIds
- * @typedef {[]} NoMissingAtxHeadingSpaceOptions
+ * @typedef {[{ checkClosedHeadings?: boolean }]} NoMissingAtxHeadingSpaceOptions
  * @typedef {MarkdownRuleDefinition<{ RuleOptions: NoMissingAtxHeadingSpaceOptions, MessageIds: NoMissingAtxHeadingSpaceMessageIds }>} NoMissingAtxHeadingSpaceRuleDefinition
  */
 
 //-----------------------------------------------------------------------------
 // Helpers
 //-----------------------------------------------------------------------------
 
-const headingPattern = /^(#{1,6})(?:[^# \t]|$)/u;
+const atxHeadingPattern = /^(#{1,6})(?:[^# \t]|$)/u;
+const atxHeadingClosedPattern = /([ \t]*)(?<!\\)(#+)([ \t]*)$/u;
```

Comment:
```suggestion
const leadingAtxHeadingHashPattern = /^(#{1,6})(?:[^# \t]|$)/u;
const trailingAtxHeadingHashPattern = /([ \t]*)(?<!\\)(#+)([ \t]*)$/u;
```

I think we could rename the variables to follow the naming convention below, or something similar, since the pre-implemented rule uses these names:

https://github.com/eslint/markdown/blob/56eb0ee4d1ffd81fe4eb7cec9dda27e35939842f/src/rules/no-duplicate-headings.js#L39-L40